app/users.py

```diff
@@ -37,6 +37,41 @@ def login():
     return render_template('login.html', title='Sign In', form=form)
 
 
+
+@bp.route('/userProfile', methods=['GET', 'POST'])
+@login_required
+def profile():
+    userInfo = UserModel.get_all_by_uid(current_user.id)
+    return render_template('userProfile.html',userInfo=userInfo)
+
+class UpdateForm(FlaskForm):
+    firstname = StringField('First Name')
+    lastname = StringField('Last Name')
+    email = StringField('Email')
+    submit = SubmitField('Update')
+
+    def validate_email(self, email):
+        if UserModel.email_exists(email.data):
+            raise ValidationError('Already a user with this email.')
+
+    
+
+@bp.route('/userUpdate', methods=['GET', 'POST'])
+@login_required
```

Comment:
This is good to know about! I didn't know that 'login_required' was a thing I could use haha